library/analytics/src/main/java/org/cru/godtools/analytics/firebase/FirebaseAnalyticsService.kt

```diff
@@ -54,38 +55,21 @@ class FirebaseAnalyticsService @VisibleForTesting internal constructor(
     @MainThread
     @Subscribe(threadMode = ThreadMode.MAIN_ORDERED)
     fun onAnalyticsEvent(event: AnalyticsBaseEvent) {
-        if (event.isForSystem(AnalyticsSystem.FIREBASE)) when (event) {
+        if (event.isForSystem(AnalyticsSystem.FIREBASE) || event.isForSystem(AnalyticsSystem.ADOBE)) when (event) {
             is AnalyticsScreenEvent -> handleScreenEvent(event)
             is AnalyticsActionEvent -> handleActionEvent(event)
         }
     }
 
-    // region ActivityLifecycleCallbacks
-    init {
-        app.registerActivityLifecycleCallbacks(this)
-    }
-
-    private var currentActivity: Activity? by weak()
-
-    override fun onActivityResumed(activity: Activity) {
-        currentActivity = activity
-    }
-
-    override fun onActivityPaused(activity: Activity) {
-        currentActivity = currentActivity?.takeUnless { it === activity }
-    }
-
-    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = Unit
-    override fun onActivityStarted(activity: Activity) = Unit
-    override fun onActivityStopped(activity: Activity) = Unit
-    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) = Unit
-    override fun onActivityDestroyed(activity: Activity) = Unit
-    // endregion ActivityLifecycleCallbacks
-    // endregion Tracking Events
-
     @MainThread
     private fun handleScreenEvent(event: AnalyticsScreenEvent) {
-        currentActivity?.let { firebase.setCurrentScreen(it, event.screen, null) }
+        val bundle = Bundle()
+        bundle.putString(FirebaseAnalytics.Param.SCREEN_NAME, event.screen)
+        bundle.putString("cru_appname", VALUE_GODTOOLS)
+        event.locale?.let { bundle.putString("cru_contentlanguage", LocaleCompat.toLanguageTag(it)) }
+        event.adobeSiteSection?.let { bundle.putString("cru_sitesection", event.adobeSiteSection) }
+        event.adobeSiteSubSection?.let { bundle.putString("cru_sitesubsection", event.adobeSiteSubSection) }
```

Comment:
another Kotlin idiom that would work here is something like:
```kotlin
val bundle = Bundle().apply {
    putString(FirebaseAnalytics.Param.SCREEN_NAME, event.screen)
    putString("cru_appname", VALUE_GODTOOLS)
    event.locale?.let { putString("cru_contentlanguage", LocaleCompat.toLanguageTag(it)) }
    event.adobeSiteSection?.let { putString("cru_sitesection", event.adobeSiteSection) }
    event.adobeSiteSubSection?.let { putString("cru_sitesubsection", event.adobeSiteSubSection) }
}
```

Reply:
this idiom makes it clear that you are configuring the bundle with everything in the `apply` block

Reply:
Got it, will go ahead and change it.